src/cpuid/src/common.rs

```diff
@@ -81,197 +70,137 @@ pub fn get_vendor_id_from_host() -> Result<[u8; 12], Error> {
-        let vendor_id = get_vendor_id_from_host();
-        assert!(vendor_id.is_ok());
-        matches!(&vendor_id.ok().unwrap(), VENDOR_ID_INTEL | VENDOR_ID_AMD);
+        };
     }
 
-    #[test]
-    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-    fn test_is_same_model() {
-        let mut curr_cpuid = CpuId::new(2).unwrap();
-
-        // Add the vendor ID leaf.
-        let vendor = get_cpuid(0x0, 0).unwrap();
-        curr_cpuid.as_mut_slice()[0].function = 0x0;
-        curr_cpuid.as_mut_slice()[0].index = 0;
-        curr_cpuid.as_mut_slice()[0].ebx = vendor.ebx;
-        curr_cpuid.as_mut_slice()[0].ecx = vendor.ecx;
-        curr_cpuid.as_mut_slice()[0].edx = vendor.edx;
-
-        // Add the feature info leaf.
-        let feature_info = get_cpuid(0x1, 0).unwrap();
-        curr_cpuid.as_mut_slice()[1].function = 0x1;
-        curr_cpuid.as_mut_slice()[1].index = 0;
-        curr_cpuid.as_mut_slice()[1].eax = feature_info.eax;
-
-        assert!(is_same_model(&curr_cpuid));
-
-        let mut diff_vendor_cpuid = curr_cpuid.clone();
-        for mut entry in diff_vendor_cpuid.as_mut_slice() {
-            if entry.function == 0x0 && entry.index == 0 {
-                entry.ebx = 0xFFFF_FFFF;
-            }
-        }
-        assert!(!is_same_model(&diff_vendor_cpuid));
-
-        let mut diff_feature_cpuid = curr_cpuid;
-        for mut entry in diff_feature_cpuid.as_mut_slice() {
-            if entry.function == 0x1 && entry.index == 0 {
-                entry.eax ^= 0x1;
-            }
-        }
-        assert!(!is_same_model(&diff_feature_cpuid));
-    }
+    // TODO: Add more dependencies.
+    cpuid_msr_dep!(
+        0x7,
+        0,
+        ebx,
+        MPX_BITINDEX,
+        [arch_gen::x86::msr_index::MSR_IA32_BNDCFGS]
+    );
+
+    // IA32_MTRR_PHYSBASEn, IA32_MTRR_PHYSMASKn
+    cpuid_msr_dep!(0x1, 0, edx, MTRR_BITINDEX, 0x200..0x210);
+
+    // Other MTRR MSRs
+    cpuid_msr_dep!(
+        0x1,
+        0,
+        edx,
+        MTRR_BITINDEX,
+        [
+            0x250, // IA32_MTRR_FIX64K_00000
+            0x258, // IA32_MTRR_FIX16K_80000
+            0x259, // IA32_MTRR_FIX16K_A0000
+            0x268, // IA32_MTRR_FIX4K_C0000
+            0x269, // IA32_MTRR_FIX4K_C8000
+            0x26a, // IA32_MTRR_FIX4K_D0000
+            0x26b, // IA32_MTRR_FIX4K_D8000
+            0x26c, // IA32_MTRR_FIX4K_E0000
+            0x26d, // IA32_MTRR_FIX4K_E8000
+            0x26e, // IA32_MTRR_FIX4K_F0000
+            0x26f, // IA32_MTRR_FIX4K_F8000
+            0x277, // IA32_PAT
+            0x2ff  // IA32_MTRR_DEF_TYPE
+        ]
+    );
+
+    // MCE MSRs
+    // We are saving 32 MCE banks here as this is the maximum number supported by KVM
+    // and configured by default.
+    // The physical number of the MCE banks depends on the CPU.
+    // The number of emulated MCE banks can be configured via KVM_X86_SETUP_MCE.
+    cpuid_msr_dep!(0x1, 0, edx, MCE_BITINDEX, 0x400..0x480);
```

Comment:
Tests that still apply to current functionality have been dropped (even test_get_vendor_id). I would like for us to make sure we add those back before merging to main.

Reply:
We discussed offline to expand test coverage in follow-up PRs on this feature branch.